pkg/ls/run_test.go

```diff
@@ -133,21 +132,37 @@ func TestRun(t *testing.T) {
 		prefix bool
 	}{
 		{
-			name: "input empty, quoted = true, long = true",
+			name: "exist, input empty, quoted = true, long = true",
 			args: []string{"ls", "-Ql"},
 			err:  nil,
 		},
 		{
-			name: "input empty, quoted = true, long = true",
+			name: "not exist, input empty, quoted = true, long = true",
 			args: []string{"ls", "-Ql", "dir"},
 			err:  os.ErrNotExist,
 		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
-			if err := run(io.Discard, tt.args); err != nil {
+			if err := Run(io.Discard, tt.args); err != nil {
 				if !errors.Is(err, tt.err) {
 					t.Errorf("list() = '%v', want: '%v'", err, tt.err)
 				}
+			} else if tt.err != nil {
```

Comment:
As far as this failing TestRun test goes, the original definition failed to check for the expected-err-got-nil case. And in fact this case could never occur because errors are not propagated since listName has only one return, return nil on [line 158](https://github.com/u-root/u-root/blob/74aa3d116bae44d273552c73e2232265d53dd4dd/cmds/core/ls/ls.go#L158).

I'm open to resolve this test issue here if we can decide what to do.